src/quarto/game/game.go

```diff
@@ -4,20 +4,36 @@ import (
 	"github.com/ahl5esoft/golang-underscore"
 )
 
-// GridSize is the Size of a grid
-const GridSize = 4
-
 // State define data for a game state
 type State struct {
-	Grid  [GridSize][GridSize]int
+	Grid  [][]int
 	Piece int
 }
 
+
+// GetNewState return a blanck state of defined size
+func GetNewState(size int) State {
+	newState := State{}
+	for i := 0; i < size; i++ {
+		newState.Grid = append(newState.Grid, []int{})
+		for j := 0; j < size; j++ {
+			newState.Grid[i] = append(newState.Grid[i], 0)
+		}
+	}
+	return newState
+}
+
+// GetGridSize return the size of a grid in a state
+func GetGridSize(state State) int {
```

Comment:
Do you really need this fonction ?
Is isn't mandatory IMO

Reply:
It's for accessing grid size without asking how a state is structured.
If it evolve, the call will be the same

Reply:
Or just for more lisibility of what is done in game package